File: 프로그래머스/Lv2/카펫.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    bool check=false;
    for(int i=brown/2;i>=1;i--){
        for(int j=1;j<=brown/2;j++){
            if(i*j==brown+yellow && yellow==i*j-2*i-(2*j-4)){
                answer.push_back(i);
                answer.push_back(j);
                check=true;
                break;
            }
        }
        if(check) break;
    }
    return answer;
}

File: 프로그래머스/Lv2/카펫.py
def solution(brown, yellow):
    answer = []
    area=brown+yellow
    
    for i in range(2,int(brown/2)): #가로 길이
        for j in range(1,i+1): #세로 길이
            if(i*j==area and (area-2*i-2*(j-2))==yellow):
                return [i,j]
    
    return answer
